### PR TITLE
feat: miner: niporep params and cbor encoding vector tests

### PR DESCRIPTION
**Pull request by rvagg**
Params implementation and tests with vectors to match https://github.com/filecoin-project/go-state-types/pull/270

### CODE REVIEW

File: builtin/v14/miner/policy.go [hunk 1]
```diff
@@ -107,6 +107,16 @@ var MaxPreCommitRandomnessLookback = builtin.EpochsInDay + ChainFinality // PARA
...
 
+// The maximum number of sector commitments in a single batch for calling ProveCommitSectorsNI.
+// Same as PreCommitSectorBatchMaxSize for consistency.
+var ProveCommitNiSectorBatchMaxSize = 256 // PARAM_SPEC
+
+// Maximum number of epochs within which to fetch a valid seal randomness from the chain for
+// a non-interactive PoRep proof. This balances the need to tie the seal to a particular chain with
+// but makes allowance for service providers to offer pre-sealed sectors within a larger window of
+// time.
+var MaxProveCommitNiLookback = abi.ChainEpoch(30*builtin.EpochsInDay) + ChainFinality // PARAM_SPEC
```
Comment (rvagg) [inline builtin/v14/miner/policy.go:118]: this will likely become 180 after a fip change

File: builtin/v14/miner/miner_types.go [hunk 2]
```diff
@@ -496,3 +496,32 @@ type SectorReturn = []PieceReturn
...
+	SealerID      abi.ActorID      // Must be set to ID of receiving actor for now
+	SealedCID     cid.Cid          // CommR
+	SectorNumber  abi.SectorNumber // unique id of sector in actor state
+	SealRandEpoch abi.ChainEpoch
+	Expiration    abi.ChainEpoch
+}
+
+// ProveCommitSectorsNIParams is the parameters for non-interactive prove committing of sectors
+// via the miner actor method ProveCommitSectorsNI.
+type ProveCommitSectorsNIParams struct {
```
Comment (rjan90) [inline builtin/v14/miner/miner_types.go:512]: We probably want to add `ProvingDeadline uint64` eventually in here as well. Given that we want to land https://github.com/filecoin-project/FIPs/blob/master/FIPS/fip-0092.md together with NI-PoRep.

But maybe worth waiting until that has moved to Last Call

File: builtin/v14/miner/policy.go (hunk 1 above)
Comment (rjan90) [reply builtin/v14/miner/policy.go:118]:
```suggestion
var MaxProveCommitNiLookback = abi.ChainEpoch(180*builtin.EpochsInDay) + ChainFinality // PARAM_SPEC
```

Comment (rvagg) [reply builtin/v14/miner/policy.go:118]: need to remove finality too

Comment (rjan90) [reply builtin/v14/miner/policy.go:118]: Ah, correct! Updated the suggested change.

Comment (rvagg) [reply builtin/v14/miner/policy.go:118]: done

File: builtin/v14/miner/policy.go [hunk 3]
```diff
@@ -107,6 +107,16 @@ var MaxPreCommitRandomnessLookback = builtin.EpochsInDay + ChainFinality // PARA
 // (2) prevents a miner attempting a long fork in the past to insert a pre-commitment after seeing the challenge.
 var PreCommitChallengeDelay = abi.ChainEpoch(150) // PARAM_SPEC
 
+// The maximum number of sector commitments in a single batch for calling ProveCommitSectorsNI.
+// Same as PreCommitSectorBatchMaxSize for consistency.
+var ProveCommitNiSectorBatchMaxSize = 256 // PARAM_SPEC
```
Comment (rvagg) [inline builtin/v14/miner/policy.go:112]: ugh, this needs to go too

**Pull request merged**